src/reanimated2/core.js

```diff
@@ -111,8 +111,18 @@ function workletValueSetter(value) {
     typeof value === 'function' ||
     (value !== null && typeof value === 'object' && value.onFrame)
   ) {
-    // animated set
     const animation = typeof value === 'function' ? value() : value;
+    // prevent setting again to the same value
+    // and triggering the mappers that treat this value as an input
+    // note that this will work for most animations
+    // this assumes that toValue === current DOES MEAN that the animation's done
+    if (
+      this._value === animation.toValue &&
+      animation.toValue === animation.current
+    ) {
+      return;
+    }
```

Comment:
What is the reason why you use here && instead of || ?

Reply:
After some checking, I decided to change that condition as I recognized 2 scenarios: either `toValue` is equal to `current`(simple animations) or `toValue` is undefined and `current` points at the target value(group animations). So I think we can skip checking `toValue` and just use `current` here(as `current` points to `toValue` until `animation.onStart` is called).

Reply:
fixed 503f327, thx for pointing out